src/components/Input/common.jsx

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright (C) 2018 The Trustees of Indiana University
+SPDX-License-Identifier: BSD-3-Clause
+*/
+import * as React from "react";
+
+import InlineAlert from "../Alert/InlineAlert";
+import * as Rivet from "../util/Rivet";
+import classNames from "classnames";
+import * as PropTypes from "prop-types";
+
+const inputClassName = (elementName, variant) => {
+  const validationVariant = variant ? `rvt-validation-${variant}` : "";
+
+  switch (elementName) {
+    case "input":
+      return classNames("rvt-text-input", validationVariant);
+    default:
+      return classNames(`rvt-${elementName}`, validationVariant);
+  }
+};
+
+const noteFragment = (id, variant, note) =>
+  variant ? (
+    <InlineAlert id={id} variant={variant}>
+      {note}
+    </InlineAlert>
+  ) : (
+    <small id={id} className="rvt-display-block">
+      {note}
+    </small>
+  );
+
+export const propTypes = {
+  /** The label for the input */
+  label: PropTypes.string.isRequired,
+  /** Visibility modifier for the input's label */
+  labelVisibility: PropTypes.oneOf(["screen-reader-only"]),
+  /** An optional note that will be displayed below the input */
+  note: PropTypes.node,
+  /** Rivet style for inline validation */
+  variant: PropTypes.oneOf(["success", "danger", "info", "warning"]),
+};
+
+export const renderInput = (
+  elementName,
+  {
+    id = Rivet.shortuid(),
+    label,
+    labelVisibility,
+    note,
+    variant,
+    className,
+    ...attrs
+  }
+) => {
+  const noteId = `${id}_note`;
+  const inputProps = {
+    id,
+    className: inputClassName(elementName, variant),
+    "aria-describedby": note ? noteId : "",
+    "aria-invalid": variant === "danger",
+    ...attrs,
+  };
+  return (
+    <div
+      className={classNames(
+        "rvt-container-sm",
+        "rvt-p-all-lg",
+        "rvt-p-all-xxl-lg-up",
+        className
+      )}
+    >
```

Comment:
Why are we wrapping the return inside this div? I think your extra spacing/centre-alignment issues in the docs are coming from this. If you remove this div, your components should not have any extra spacing.

Reply:
This makes so much more sense. I copied the styleguide markup. Woops! I also understand now why the original markup documentation sample had margins. I added those back. I do still have the div because we use that as a pass through for custom className. I hope that makes sense, I got it from 1.0: https://github.com/indiana-university/rivet-react/blob/71e2da570662f15829989c2f9bce7562bd0c18c9/src/components/Input/common.tsx#L44

It does look like `font-weight: 700;` was removed in rivet 2.0 from rvt-label. But unless I'm crossing more wires it looks like that was an intentional design choice.

Thanks so much! Let me know what you think on the last revision. It looks so much better now.